File: compiler-rt/lib/ubsan/lit_tests/Misc/unreachable.cpp
// RUN: %clang -fsanitize=unreachable %s -O3 -o %t && %t 2>&1 | FileCheck %s

int main(int, char **argv) {
  // CHECK: unreachable.cpp:5:3: runtime error: execution reached a __builtin_unreachable() call
  __builtin_unreachable();
}

File: llgo/test/execution/functions/unreachable.go
// RUN: llgo -o %t %s
// RUN: %t 2>&1 | FileCheck %s

// CHECK: f1
// CHECK-NEXT: f2
// CHECK-NEXT: f3
// CHECK-NEXT: f4
// CHECK-NEXT: 123

package main

func f1() {
	if true {
		println("f1")
		return
	}
	for {
	}
}

func f2() {
	defer func() { println("f2") }()
	if true {
		return
	}
	for {
	}
}

func f3() int {
	if true {
		println("f3")
		return 123
	}
	for {
	}
}

func f4() int {
	defer func() { println("f4") }()
	if true {
		return 123
	}
	for {
	}
}

func main() {
	f1()
	f2()
	f3()
	println(f4())
}
